### PR TITLE
Ltd 4170 sub status UI

### PR DESCRIPTION
**Pull request by kevincarrogan**
### Aim

Add a view to allow sub-statuses to be edited.

At the moment there is no link to this view, I'm thinking that we should add this link once we add the sub-status onto the product details page.

[LTD-4170](https://uktrade.atlassian.net/browse/LTD-4170)

[LTD-4170]: https://uktrade.atlassian.net/browse/LTD-4170?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: caseworker/cases/services.py [hunk 1]
```diff
@@ -439,3 +439,19 @@ def post_compliance_person_present(request, case_id, json):
 
         return {**json, "errors": translated_errors}, data.status_code
     return data.json(), data.status_code
+
+
+def get_case_sub_statuses(request, case_id):
+    response = client.get(request, f"/applications/{case_id}/sub-statuses/")
```
Comment (depsiatwal) [inline caseworker/cases/services.py:445]: Does this end point exist is it part of another PR ?

File: caseworker/cases/services.py [hunk 2]
```diff
@@ -439,3 +439,19 @@ def post_compliance_person_present(request, case_id, json):
...
+    response.raise_for_status()
+    return response.json()
+
+
+def put_case_sub_status(request, case_id, data):
+    response = client.put(
+        request,
+        f"/applications/{case_id}/sub-status/",
```
Comment (depsiatwal) [inline caseworker/cases/services.py:453]: same as above

File: caseworker/templates/case/form.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% extends 'layouts/case.html' %}
```
Comment (depsiatwal) [inline caseworker/templates/case/form.html:1]: do we need another form or can we just use core form ?

**Review by depsiatwal — COMMENTED** [review]
lgtm just some minor comments more for my info

File: caseworker/cases/services.py (hunk 1 above)
Comment (kevincarrogan) [reply caseworker/cases/services.py:445]: It exists. It's on the API side so can't be a part of this PR.

https://github.com/uktrade/lite-api/pull/1578

It's already merged in.

File: caseworker/templates/case/form.html (hunk 3 above)
Comment (kevincarrogan) [reply caseworker/templates/case/form.html:1]: Core form doesn't have the case navigation header. This uses an include to make sure that we aren't duplicating the form display logic though.

File: caseworker/cases/views/main.py [hunk 4]
```diff
@@ -392,6 +401,75 @@ def get_success_url(self):
...
+    def form_valid(self, form):
+        self.put_case_sub_status(
+            self.request,
+            self.case.id,
+            form.cleaned_data,
+        )
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return reverse(
```
Comment (currycoder) [inline caseworker/cases/views/main.py:463]: Is a success message injected in some way?  I think I always forget this..

**Review by currycoder — APPROVED** [review]

File: caseworker/cases/views/main.py (hunk 4 above)
Comment (kevincarrogan) [reply caseworker/cases/views/main.py:463]: Not automatically. I did forget that in this case, I will add it in.

**Review by depsiatwal — APPROVED** [review]
lgtm

**Pull request merged**